src/Translator.php

```diff
@@ -55,22 +55,19 @@ public function run()
         $this->outputFormatter = new OutputFormatter(true);
         $this->checkEnv();
         $this->setModulePaths();
+        $this->log('Updating translations for ' . count($this->modulePaths) . ' module(s)');
         if ($this->doTransifexPullAndUpdate) {
-            $this->log('Updating translations for ' . count($this->modulePaths) . ' module(s)');
-            $this->storeJson();
-            $this->storeYaml();
+            $this->log('Starting transifex pull block');
+            $this->backupContent();
             $this->setJsonAndYmlFileTimes();
             $this->transifexPullSource();
-            $this->mergeYaml();
-            $this->removeEnglishStringsFromYamlTranslations();
-            $this->cleanYaml();
-            $this->mergeJson();
-            $this->removeEnglishStringsFromJsonTranslations();
-            $this->removeEmptyYamlFiles();
-            $this->removeEmptyJsFiles();
+            $this->mergeAndCleanContent();
         }
         if ($this->doCollectStrings) {
+            $this->log('Starting string collection block');
+            $this->backupContent();
             $this->collectStrings();
+            $this->mergeAndCleanContent();
```

Comment:
While the text collector doesn't currently deal with JS localisation strings, I've chosen to include JS in the backup and restore here because:

1. I can absolutely see us adding that functionality in the future - having this here now means we don't even have to worry about it breaking this tool if/when we do so
2. Having this do the JS doesn't cause any problems - it'll just effectively copy the content into memory and then put it back into the files as-is in the worst case (effectively a no-op).